### PR TITLE
[PM-5024] Migrate tax-info component

### PR DESCRIPTION
**Pull request by cyprain-okeke**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-5024
## 📔 Objective

Migrate tax information

## 📸 Screenshots

Screenshot 2024-06-28 at 12 23 43
Screenshot 2024-06-28 at 12 24 42

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: apps/web/src/app/billing/accounts/trial-initiation/trial-billing-step.component.ts [hunk 1]
```diff
@@ -86,6 +86,10 @@ export class TrialBillingStepComponent implements OnInit {
   }
 
   async submit(): Promise<void> {
+    if (!this.taxInfoComponent.taxFormGroup.valid) {
+      this.taxInfoComponent.taxFormGroup.markAllAsTouched();
```
Comment (amorask-bitwarden) [inline apps/web/src/app/billing/accounts/trial-initiation/trial-billing-step.component.ts:90]: ❓ Are we supposed to early return when the form is invalid?

File: apps/web/src/app/billing/individual/premium.component.ts [hunk 2]
```diff
@@ -65,6 +65,9 @@ export class PremiumComponent implements OnInit {
     }
   }
   submit = async () => {
+    if (!this.taxInfoComponent.taxFormGroup.valid) {
+      this.taxInfoComponent.taxFormGroup.markAllAsTouched();
```
Comment (amorask-bitwarden) [inline apps/web/src/app/billing/individual/premium.component.ts:69]: ❓ Same question here.

File: apps/web/src/app/billing/organizations/organization-plans.component.ts [hunk 3]
```diff
@@ -549,6 +549,10 @@ export class OrganizationPlansComponent implements OnInit, OnDestroy {
   }
 
   submit = async () => {
+    if (!this.taxComponent?.taxFormGroup.valid) {
+      this.taxComponent?.taxFormGroup.markAllAsTouched();
```
Comment (amorask-bitwarden) [inline apps/web/src/app/billing/organizations/organization-plans.component.ts:553]: ❓ Same question here.

**Review by amorask-bitwarden — COMMENTED** [review]
Just one question on form validity.

File: apps/web/src/app/billing/accounts/trial-initiation/trial-billing-step.component.ts (hunk 1 above)
Comment (cyprain-okeke) [reply apps/web/src/app/billing/accounts/trial-initiation/trial-billing-step.component.ts:90]: yeah, it should return. done

**Pull request merged**